ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,266 @@
+from opentelemetry.util import types
+
+from opentelemetry.ext.cloud_trace.version import __version__
+
+logger = logging.getLogger(__name__)
+
+AGENT = "opentelemetry-python [{}]".format(__version__)
+# Max length is 128 bytes for a truncatable string.
+MAX_LENGTH = 128
+
+
+class CloudTraceSpanExporter(SpanExporter):
+    """Cloud Trace span exporter for OpenTelemetry.
+
+    Args:
+        client: Cloud Trace client.
+        project_id: project_id to create the Trace client.
+    """
+
+    def __init__(
+            self, client=None, project_id=None,
+    ):
+        if client is None:
+            client = TraceServiceClient()
+        self.client = client
+        self.project_id = project_id
+
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        """Export the spans to Cloud Trace.
+
+        See: https://cloud.google.com/trace/docs/reference/v2/rest/v2/
+             projects.traces/batchWrite
+
+        Args:
+            spans: Tuple of spans to export
+        """
+        cloud_trace_formatted_spans = self.translate_to_cloud_trace(spans)
+        cloud_trace_spans = []
+        for span in cloud_trace_formatted_spans:
+            try:
+                cloud_trace_spans.append(self.client.create_span(**span))
+            except Exception as ex:
+                logger.warning("Error {} when creating span {}".format(ex, span))
+
+        try:
+            self.client.batch_write_spans(
+                "projects/{}".format(self.project_id),
+                cloud_trace_spans,
+            )
+        except Exception as ex:
+            logger.warning("Error while writing to Cloud Trace: %s", ex)
+            return SpanExportResult.FAILED_RETRYABLE
+
+        return SpanExportResult.SUCCESS
+
+    def translate_to_cloud_trace(
+            self, spans: Sequence[Span]
+    ) -> List[Dict[str, Any]]:
+        """Translate the spans to Cloud Trace format.
+
+        Args:
+            spans: Tuple of spans to convert
+        """
+
+        cloud_trace_spans = []
+
+        for span in spans:
+            ctx = span.get_context()
+            trace_id = trace_api.format_trace_id(ctx.trace_id)[2:]
+            span_id = trace_api.format_span_id(ctx.span_id)[2:]
+            span_name = "projects/{}/traces/{}/spans/{}".format(
+                self.project_id, trace_id, span_id
+            )
+
+            parent_id = None
+            if isinstance(span.parent, trace_api.Span):
+                parent_id = trace_api.format_span_id(span.parent.get_context().span_id)[2:]
+            elif isinstance(span.parent, trace_api.SpanContext):
+                parent_id = trace_api.format_span_id(span.parent.span_id)[2:]
+
+            start_time = get_time_from_ns(span.start_time)
+            end_time = get_time_from_ns(span.end_time)
+
+            attributes = _extract_attributes(span.attributes)
+            attributes['attribute_map']["g.co/agent"] = _format_attribute_value(AGENT)
+
+            sd_span = {
```

Comment:
I assume "sd" stands for Stackdriver? We should give this variable a better name since the Stackdriver name is no longer in use.

Or better yet, since it's only used once we can probably avoid assigning it and just have something like
```
cloud_trace_spans.append({
  "name": span_name,
  ...
})
```